### PR TITLE
Change type alias to creating a new type

### PR DESCRIPTION
**Pull request by bowei**
Fixes the documentation generation issue.

### CODE REVIEW

**Review by hbagdi — APPROVED** [review]

File: apis/v1alpha1/gatewayclass_types.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ type GatewayClassSpec struct {
 // resource for a GatewayClass.
 //
 // +k8s:deepcopy-gen=false
-type GatewayClassParametersObjectReference = LocalObjectReference
+type GatewayClassParametersObjectReference LocalObjectReference
```
Comment (robscott) [inline apis/v1alpha1/gatewayclass_types.go:106]: Is there any point in leaving these aliases in place or should we just use `LocalObjectReference` directly? I think we're already doing that for a lot of references in the API.

Comment (jpeach) [reply apis/v1alpha1/gatewayclass_types.go:106]: Yeh, just using `LocalObjectReference` directly makes it easier to read IMHO

Comment (bowei) [reply apis/v1alpha1/gatewayclass_types.go:106]: Don't really have a preference, I can change it.

**Comment by bowei** [comment]
Updated

**Pull request merged**